Mathlib/Data/ZMod/Basic.lean

```diff
@@ -772,6 +772,9 @@ theorem inv_mul_of_unit {n : ℕ} (a : ZMod n) (h : IsUnit a) : a⁻¹ * a = 1 :
   rw [mul_comm, mul_inv_of_unit a h]
 #align zmod.inv_mul_of_unit ZMod.inv_mul_of_unit
 
+theorem inv_eq_of_mul_eq_one (n : ℕ) (a b : ZMod n) (h : a * b = 1) : a⁻¹ = b :=
```

Comment:
```suggestion
-- TODO: If we changed `⁻¹` so that `ZMod n` is always a `DivisionMonoid`,
-- then we could use the general lemma `inv_eq_of_mul_eq_one`
protected theorem inv_eq_of_mul_eq_one (n : ℕ) (a b : ZMod n) (h : a * b = 1) : a⁻¹ = b :=
```

Reply:
@YaelDillies what needs to be done to make `ZMod n` a division monoid? Is it an invasive change?

Reply:
Not so invasive, since that only involves changing the junk values of `⁻¹`. Currently, `a * a⁻¹ = gcd a n`. We could change it so that `a⁻¹` is the proper inverse when it exists and `a` when it doesn't (this respects the `DivisionMonoid` axioms).

However @kbuzzard complained that this wouldn't work very well with the Chinese Remainder theorem, namely `(a⁻¹, b⁻¹) ≠ (a, b)⁻¹` in `ZMod (p * q) ≃ ZMod p × ZMod q` (AFAICT this is already not the case with the current definition).

So a solution would be to define `a⁻¹` on `ZMod (p ^ n)` by the proper inverse if it exists and `a` otherwise, and then extend that to `ZMod n` for all `n` by CRT. However that would mean pulling (some special case of) CRT in for the definition, which is a bit annoying.

Reply:
But if we take that CRT approach, then we still don't have a `DivisionMonoid`... which would also be sad.

Reply:
Why? Every `ZMod p` would be a `DivisionMonoid` (by my first argument), hence by CRT every `ZMod n` would be too (because products of division monoids are division monoids, and being a division monoid is preserved under isomorphism).

Reply:
Nevermind, I was confused. Thanks for clarifying!